stack/corepersistence/token/src/test/java/org/apache/usergrid/persistence/token/TokenSerializationTest.java

```diff
@@ -0,0 +1,202 @@
+
+        UUID uuid = UUIDGenerator.newTimeUUID();
+        Map<String, Object> tokenDetails = new HashMap<>();
+        tokenDetails.put(TokenSerializationImpl.TOKEN_TYPE, "test-token");
+
+        long accessedTime = System.currentTimeMillis()-1000000;
+        long inactiveTime = accessedTime+2000000;
+
+        tokenDetails.put(TokenSerializationImpl.TOKEN_ACCESSED, accessedTime);
+        tokenDetails.put(TokenSerializationImpl.TOKEN_INACTIVE, inactiveTime);
+
+        ByteBuffer principalKeyBuffer = ByteBuffer.wrap("test-principal".getBytes());
+
+        tokenSerialization.putTokenInfo(uuid, tokenDetails, principalKeyBuffer, 60);
+
+        Map<String, Object> returnedDetails = tokenSerialization.getTokenInfo(uuid);
+
+        assertEquals(
+            accessedTime,
+            returnedDetails.get(TokenSerializationImpl.TOKEN_ACCESSED)
+        );
+
+        long newAccessedTime = System.currentTimeMillis();
+        long newInactiveTime = newAccessedTime+1000000;
+
+        tokenSerialization.updateTokenAccessTime(uuid, newAccessedTime,newInactiveTime, 1200);
+
+        returnedDetails = tokenSerialization.getTokenInfo(uuid);
+
+        assertEquals(
+            newAccessedTime,
+            returnedDetails.get(TokenSerializationImpl.TOKEN_ACCESSED)
+        );
+    }
+
+    @Test
+    public void deleteTokens() {
+
+        UUID uuid = UUIDGenerator.newTimeUUID();
+        Map<String, Object> tokenDetails = new HashMap<>();
+        tokenDetails.put(TokenSerializationImpl.TOKEN_TYPE, "test-token");
+
+        ByteBuffer principalKeyBuffer = ByteBuffer.wrap("test-principal".getBytes());
+
+        tokenSerialization.putTokenInfo(uuid, tokenDetails, principalKeyBuffer, 60);
+
+        Map<String, Object> returnedDetails = tokenSerialization.getTokenInfo(uuid);
+
+        assertEquals(
+            tokenDetails.get(TokenSerializationImpl.TOKEN_TYPE),
+            returnedDetails.get(TokenSerializationImpl.TOKEN_TYPE)
+        );
+
+        tokenSerialization.deleteTokens(Collections.singletonList(uuid), principalKeyBuffer);
+
+        returnedDetails = tokenSerialization.getTokenInfo(uuid);
+
+        assertEquals(
+            0,
+            returnedDetails.size()
+        );
+    }
+
+    @Test
+    public void revokeToken() {
+
+        UUID uuid = UUIDGenerator.newTimeUUID();
+        Map<String, Object> tokenDetails = new HashMap<>();
+        tokenDetails.put(TokenSerializationImpl.TOKEN_TYPE, "test-token");
+
+        ByteBuffer principalKeyBuffer = ByteBuffer.wrap("test-principal".getBytes());
+
+        tokenSerialization.putTokenInfo(uuid, tokenDetails, principalKeyBuffer, 60);
+
+        Map<String, Object> returnedDetails = tokenSerialization.getTokenInfo(uuid);
+
+        assertEquals(
+            tokenDetails.get(TokenSerializationImpl.TOKEN_TYPE),
+            returnedDetails.get(TokenSerializationImpl.TOKEN_TYPE)
+        );
+
+        tokenSerialization.revokeToken(uuid, principalKeyBuffer);
```

Comment:
Do you want to revokeToken with a null principalKeyBuffer too, since that is a use case? I assume that if the principal token info has not been deleted during the revokeToken, getting tokens for the principal will prune that token? Anything to test there?